lib/parallel_tests.rb

```diff
@@ -25,6 +20,18 @@ def determine_number_of_processes(count)
       ].detect{|c| not c.to_s.strip.empty? }.to_i
     end
 
+    def pids
+      @pids ||= Pids.new(pid_file_path)
+    end
+
+    def pid_file_path
+      ENV['PARALLEL_PID_FILE'] ||= new_pid_file_path
+    end
+
+    def new_pid_file_path
```

Comment:
why not `ENV ||= Tempfile.new ... ` ?

Reply:
We needed a separate method to set the memoized variable so we don't generate a new tempfile every time it's called. `pid_file_path` is being called by the runner to set then env variable

Reply:
but won't it be in the ENV then ... so the `||=` will prevent the new_pid_file_path from getting called ? ... nobody calls new_pid_file_path directly anyway

Reply:
Oh ha yup, forgot I was doing `||=` there